### PR TITLE
[TTS] Update UnivNet, HiFi-GAN and WaveGlow, small fixes in Mixer-TTS, FastPitch and Exportable

### PR DESCRIPTION
**Pull request by Oktai15**
**Collection**: [tts]

# Changelog 
- refactoring of HiFi-GAN code and config, replacing deprecated datasets with `VocoderDataset`
- refactoring of UnivNet code and config, replacing deprecated datasets with `VocoderDataset`, add exportability (it works correctly only for TorchScript)
- refactoring of WaveGlow code and config, replacing deprecated datasets with `VocoderDataset`
- minor refactoring of FastPitch code 
- minor refactoring of Mixer-TTS code 
- fixing small bugs in Exportable
  
**PR Type**:
- [x] New Feature
- [x] Bugfix

### CODE REVIEW

File: nemo/core/classes/exportable.py [hunk 1]
```diff
@@ -192,7 +192,6 @@ def export(
                             self,
                             {"forward": tuple(input_list) + tuple(input_dict.values())},
                             strict=False,
-                            optimize=True,
```
Comment (blisc) [inline nemo/core/classes/exportable.py:195]: Why remove this line?

**Review by blisc — CHANGES_REQUESTED** [review]

File: examples/tts/conf/hifigan/model/generator/v1.yaml [hunk 2]
```diff
@@ -1,4 +1,3 @@
-# @package _group_
```
[the remark this answers is not in the document]
Comment (Oktai15) [reply examples/tts/conf/hifigan/model/generator/v1.yaml:1]: Yes, I know. We needed it for hydra-core=1.0.0. Now we support only hydra-core>=1.1.0. Check this article: https://hydra.cc/docs/next/upgrades/1.0_to_1.1/changes_to_package_header.

File: nemo/collections/tts/models/hifigan.py [hunk 3]
```diff
@@ -397,25 +388,34 @@ def load_state_dict(self, state_dict, strict=True):
...
+            "audio": NeuralType(('B', 'S', 'T'), AudioSignal(self.sample_rate)),
+        }
+
     def input_example(self, max_batch=1, max_dim=256):
         """
         Generates input examples for tracing etc.
         Returns:
             A tuple of input examples.
         """
         par = next(self.parameters())
```
[the remark this answers is not in the document]
Comment (Oktai15) [reply nemo/collections/tts/models/hifigan.py:417]: We need it to get correct dtype (fp16 vs fp32).

File: nemo/core/classes/exportable.py [hunk 4]
```diff
@@ -182,7 +182,7 @@ def export(
                 jitted_model = None
                 if try_script:
                     try:
-                        jitted_model = torch.jit.script(module)
```
[the remark this answers is not in the document]
Comment (Oktai15) [reply nemo/core/classes/exportable.py:185]: I don't know why, but `module` variable is not declared, so I fix it.

File: nemo/collections/tts/modules/univnet_modules.py [hunk 5]
```diff
@@ -259,7 +259,7 @@ def location_variable_convolution(self, x, kernel, bias, dilation=1, hop_size=25
         '''
         batch, _, in_length = x.shape
         batch, _, out_channels, kernel_size, kernel_length = kernel.shape
-        assert in_length == (kernel_length * hop_size), "length of (x, kernel) is not matched"
+        # in_length == (kernel_length * hop_size)
```
[the remark this answers is not in the document]
Comment (Oktai15) [reply nemo/collections/tts/modules/univnet_modules.py:262]: It continues silently. Reverted.

File: examples/tts/conf/hifigan/model/train_ds/train_ds.yaml [hunk 6]
```diff
@@ -1,11 +1,10 @@
-# @package _group_
 dataset:
-  _target_: "nemo.collections.tts.data.datalayers.AudioDataset"
+  _target_: "nemo.collections.tts.torch.data.VocoderDataset"
   manifest_filepath: ${train_dataset}
+  sample_rate: ${sample_rate}
+  n_segments: 8192
   max_duration: null
   min_duration: 0.75
```
[the remark this answers is not in the document]
Comment (Oktai15) [reply examples/tts/conf/hifigan/model/train_ds/train_ds.yaml:7]: Agree, will fix it.

File: nemo/core/classes/exportable.py (hunk 1 above)
Comment (blisc) [reply nemo/core/classes/exportable.py:195]: Are we using `torch.jit.optimized_execution()`?
I don't think any models currently use TORCHSCRIPT. Let's leave exportable.py untouched and we'll fix it if and when errors arise in the future.

**Pull request merged**